frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/ReferenceTab/ReferenceTab.test.tsx

```diff
@@ -1,29 +1,38 @@
 import React from 'react';
-import { screen } from '@testing-library/react';
+import { screen, waitForElementToBeRemoved } from '@testing-library/react';
 import { ReferenceTab } from './ReferenceTab';
 import { renderWithProviders } from '../../../../../../testing/mocks';
 import { ComponentType } from 'app-shared/types/ComponentType';
 import type { FormComponent } from '../../../../../../types/FormComponent';
 import { textMock } from '@studio/testing/mocks/i18nMock';
 import userEvent from '@testing-library/user-event';
 import { componentMocks } from '../../../../../../testing/componentMocks';
+import { queriesMock } from 'app-shared/mocks/queriesMock';
 
 const mockComponent = componentMocks[ComponentType.Dropdown];
+const mockOptionListIds = jest
+  .fn()
+  .mockImplementation(() => Promise.resolve<string[]>(['test1', 'test2']));
 
 describe('ReferenceTab', () => {
-  it('should render', () => {
+  it('should render', async () => {
     renderReferenceTab();
+    await waitForElementToBeRemoved(() =>
+      screen.queryByText(textMock('ux_editor.modal_properties_loading')),
+    );
```

Comment:
If we are not testing that the spinner is in the document, which i think is not necessary here, we can put the data on the cache. But I'll leave that decision up to you.